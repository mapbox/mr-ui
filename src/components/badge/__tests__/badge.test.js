import React from 'react';
import { shallow } from 'enzyme';
import { testCases } from './badge-test-cases';

describe('badge', () => {
  describe(testCases.basic.description, () => {
    const wrapper = shallow(
      React.createElement(testCases.basic.component, testCases.basic.props)
    );

    test('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('no Tooltip', () => {
      expect(wrapper.find('Tooltip').exists()).toBe(false);
    });
  });

  describe(testCases.color.description, () => {
    const wrapper = shallow(
      React.createElement(testCases.color.component, testCases.color.props)
    );

    test('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('no Tooltip', () => {
      expect(wrapper.find('Tooltip').exists()).toBe(false);
    });
  });

  describe(testCases.tooltip.description, () => {
    const wrapper = shallow(
      React.createElement(testCases.tooltip.component, testCases.tooltip.props)
    );

    test('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('tooltipText renders Tooltip', () => {
      expect(wrapper.find('Tooltip').exists()).toBe(true);
    });
  });

  describe(testCases.colorAndTooltip.description, () => {
    const wrapper = shallow(
      React.createElement(
        testCases.colorAndTooltip.component,
        testCases.colorAndTooltip.props
      )
    );

    test('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('tooltipText renders Tooltip', () => {
      expect(wrapper.find('Tooltip').exists()).toBe(true);
    });
  });
});
