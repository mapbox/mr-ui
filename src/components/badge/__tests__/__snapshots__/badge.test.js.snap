// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`badge basic renders as expected 1`] = `
<div
  className="txt-s txt-bold round px6 py3 inline-block cursor-default bg-orange-faint color-orange-dark "
>
  Hello
</div>
`;

exports[`badge color and tooltip renders as expected 1`] = `
<Tooltip
  alignment="center"
  block={false}
  coloring="light"
  content="A call used to greet someone or draw attention to something from a distance."
  disabled={false}
  maxWidth="small"
  padding="small"
  placement="top"
  respondsToClick={false}
  textSize="s"
>
  <div
    className="txt-s txt-bold round px6 py3 inline-block cursor-default bg-purple-faint color-purple "
  >
    Ahoy
  </div>
</Tooltip>
`;

exports[`badge color renders as expected 1`] = `
<div
  className="txt-s txt-bold round px6 py3 inline-block cursor-default bg-blue-faint color-blue "
>
  Blue
</div>
`;

exports[`badge tooltip renders as expected 1`] = `
<Tooltip
  alignment="center"
  block={false}
  coloring="light"
  content="The best day."
  disabled={false}
  maxWidth="small"
  padding="small"
  placement="top"
  respondsToClick={false}
  textSize="s"
>
  <div
    className="txt-s txt-bold round px6 py3 inline-block cursor-default bg-orange-faint color-orange-dark "
  >
    Friday
  </div>
</Tooltip>
`;
